lib/auth/grpcserver.go

```diff
@@ -2163,11 +2170,13 @@ func downgradeRoleToV6(r *types.RoleV6) (*types.RoleV6, bool, error) {
 	case types.V3, types.V4, types.V5, types.V6:
 		return r, false, nil
 	case types.V7:
-		var (
-			downgraded types.RoleV6
-			restricted bool
-		)
-		downgraded = *r
+		var restricted bool
+		// Make a shallow copy of the role so that we don't mutate the original.
+		// This is necessary because the role is stored in the backend and it's shared
+		// between multiple clients sessions.
```

Comment:
I don't think it's really related to the role being stored in the backend, which I'd think of as the database, but I'm not 100% sure where the shared reference actually comes from, probably the watcher or maybe the cache? Could probably just remove this section of the comment